### DISCUSSION TITLE
Use pydantic to exclude/redact specific keys in model and sub-models of response data

### DISCUSSION THREAD

**Discussion by mikelane** [pydantic/pydantic#4770] [Question]
We are in need of redacting data from certain keys from a `requests.Response.json()` dict. We don't know what model will be represented in the request or response, but the model and/or its submodels may very often have a client_id and client_secret. I was hoping I could do something like this:

```python
class RedactedResponse(BaseModel, extra=Extra.allow):
    class Config:
        extra = Extra.allow
        fields = {
            '__all__': {
                'client_id': {'exclude': True}, 
                'client_secret': {'exclude': True}
            }
        }

response = requests.get(url)
redacted_response_data = RedactedResponse.parse_obj(response.json())
# redacted_response_data.json() would not include client_id or client_secret in the root
#  or in any nested models that may exist.
```

I see in the docs that the `__all__` keyword is for any nested list or tuples, not nested models, per se. I also see that there is a way to specify the excluded keys from a nested model, but this would require knowing the overall model structure, which we wouldn't as it stands.

Just wondering if there is a way to go about this that I've not been able to figure out?

**Comment by samuelcolvin** [comment]
Have you looked at `exclude`? See https://pydantic-docs.helpmanual.io/usage/exporting_models/#model-and-field-level-include-and-exclude

**Reply by mikelane** [reply]
Thanks for the response. I've definitely looked at exclude. It doesn't appear to do what I need, so I'm just checking to see if there is a way to get what I need from the existing functionality.

**Comment by samuelcolvin** [comment]
I'm afraid you'll need your own logic for this. Probably best to implement a custom `dict` method, call super, then customise the result.
